### ISSUE TITLE
Enforce style of passing an event or part of an event to a method

### ISSUE THREAD

**Issue by leonheess** [vuejs/eslint-plugin-vue#1611]
**Please describe what the rule should do:**
In Vue you can use `test($event.detail)` as well as  `event => test(event.detail)` to achieve the same thing. 1. should be preferred.

1. `test($event.detail)`
```vue
<Component
  @click="test($event.detail)"
/>
```

2. `event => test(event.detail)`
```vue
<Component
  @click="event => test(event.detail)"
/>
```

**What category should the rule belong to?**
[X] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<Component
  @click="event => test(event.detail)"
/>
```

```vue
<Component
  @click="({ detail }) => test(detail)"
/>
```

**Comment by ota-meshi** [comment]
Thank you for the rule proposal!
I think it would be even better if the options allowed the user to choose the style they prefer.

**Comment by FloEdelmann** [comment]
Instead of a separate rule, it could maybe be a new option for [vue/v-on-function-call](https://eslint.vuejs.org/rules/v-on-function-call.html).

**Comment by FloEdelmann** [comment]
This was implemented as the new [`vue/v-on-handler-style`](https://eslint.vuejs.org/rules/v-on-handler-style.html) rule in #2009.

**Comment by leonheess** [comment]
Doesn't seem to work:

`'vue/v-on-handler-style': ['error', ['method', 'inline']]`

![image](https://github.com/vuejs/eslint-plugin-vue/assets/13834296/65b93a6b-4a6c-448e-b9bd-97b1eac251a6)

**Comment by ota-meshi** [comment]
I think you should use the following option:

`'vue/v-on-handler-style': ['error', 'inline']`

https://eslint.vuejs.org/rules/v-on-handler-style.html#inline

**Comment by leonheess** [comment]
@ota-meshi But then it will complain about `@click="function"` 😞

**Comment by ota-meshi** [comment]
I'm not sure. I thought you prefer the inline format. Isn't it?

**Comment by leonheess** [comment]
@ota-meshi I want it to be as short as possible so if no parentheses are needed, none should be used.

**Comment by ota-meshi** [comment]
I'm not sure which style you prefer. Could you please open a new issue as a feature request for the v-on-handler-style rule and detail your preferred style?
